Fix for DOC-450, 451, 452, 453 (#268)
Macrometacorp/docs: docs/cep/reference/functions/trigger/trigger.md

---
title: Cron Trigger
---

> Triggers allow events to be periodically generated. Trigger definition can be used to define a trigger. A trigger also works like a stream with a predefined schema.

Reviewer: Great intro! I suggest linking the word "stream" to the Streams section of the documentation.

Reviewer: Is it "trigger" or "cron trigger"? The title does not agree with the text.

## Features

> 

Reviewer: ```suggestion
```

For some use cases the system should be able to periodically generate events based on a specified time interval to perform some periodic executions.

A trigger can be performed for a 'start' operation, for a given <time interval>, or for a given '<cron expression>'.

> The following types of triggeres are currently supported:

Reviewer: ```suggestion
## Trigger Types

The following types of triggers are currently supported:
```

> | Trigger type           | Description                                                                                                        |	

Reviewer: ```suggestion
| Trigger Type           | Description                                                                                                        |	
```

|------------------------|--------------------------------------------------------------------------------------------------------------------|
| every <time interval>  | An event is triggered periodically at the given time interval (Minimum supported time interval is millisec)        |
| '<cron expression>'    | An event is triggered periodically based on the given cron expression                                              |
| 'start'                | An event is triggered when Siddhi is started                                                                       |

## Syntax

Trigger uses the following syntax:

```js
	CREATE TRIGGER <trigger name> WITH ( interval = <time interval> );
	CREATE TRIGGER <trigger name> WITH ( expression = 'start' or '<cron expression>' ); 
```

> 

Reviewer: ```suggestion
```

## Example

> 

Reviewer: Need an intro here. Also, why are there numbers before the commands in the code block? Will this block run if the user copies and pastes it with those numbers there?

```js

> -- A trigger to generate events every 5 seconds.

Reviewer: ```suggestion
-- A trigger to generate events every five seconds.
```
